### ISSUE TITLE
Have a signedIdentity type to separate physical and logical parts

### ISSUE THREAD

**Issue by sjhx** [containers/image#884]
Related to https://github.com/containers/image/issues/807
Taking on board the comments in the above issue, perhaps a more acceptable proposal
would be to allow the signedIdentity to be matched "on the right" where the logical part 
of the identity is maintained.

This would allow the physical image to be moved but the logical part of the image reference to be confirmed. 

For example you could move:
`icr.io/product/ibm/db2/manager:9.1.1.1`-> `myreg.com/myspace/prod/ibm/db2/manager:9.1.1.1`

and with something like:
```
    signedIdentity:
        type: "matchLogicalAndTag"
        folders: 3 
```
or 
```
    signedIdentity: 
        type: "imageMirror"
        signedPhysicalPrefix: "icr.io/product"
        matchTag: true
```

still be reasonable assured that the image contains the wanted asset

**Comment by mtrmac** [comment]
Instead of the very inflexible `exactReference` or the imprecise, tag-ignoring `exactRepository`, I’ve been considering something like:

```yaml
signedIdentity:
  type: "remapIdentity"
  prefix: "myreg/com/myspace/prod/ibm"
  signedPrefix: "irc.rio/product/ibm"
```
(where the prefix must be a complete namespace/repository, up to a `/:@` separator, and is remapped, but the rest must match exactly, probably hard-coding the `matchRepoDigestOrExact` semantics, or maybe making the `matchExact`/`matchRepoDigestOrExact`/`matchRepository` choice orthogonal to the remapping).

The idea is that the registry of a vendor would be mirrored in a way that broadly preserves the vendor’s repository structure, and there would ideally be a single `policy.json` scope for the vendor, with a single set of public keys and a single remapping configuration.

Would that work for you?

The `prefix` part, in the example above, somewhat redundant with the policy scope name (the key in the `docker` transport map), but decoupling them gives more flexibility to the rewrites, or possibly it could be an optional field that defaults to the policy scope name.

---

I think this is basically similar to the `imageMirror` example (where I’m not sure what `matchTag` does).

The `folders: 3` match-the-right-part functionality is conceptually similar but intuitively seems too loose to me:
- It doesn’t document the expected source/destination repo layout, making it difficult to review/update.
- It would be more difficult to reason about what kinds of repositories match, and to track what that means — e.g. if a vendor creates a beta-users registry with a different hostname, suddenly such images would be accepted if they are signed by the same keys.

**Comment by sjhx** [comment]
I believe it works.

I agree that using a number of folders is too loose and hard to reason. Precisely defining the original physical location of the image is precise and has the advantages indicated. 

You said that the `prefix` part in the example is possibly redundant, I think it is, it seems like it can only be right and could be computed or wrong and always fail the policy or am I missing something?

Are you say this requirement would be in addition to other policy requirements whiich would determine if the tag was matched or not?

**Comment by mtrmac** [comment]
`remapIdentity` was implemented in #1075 . The `prefix` is explicit because the syntax of the “scope” (the key in the map of policy requirement arrays) is not necessarily using the same syntax for non-`docker` transports.
